packages/block-editor/src/store/selectors.js

```diff
@@ -464,6 +465,33 @@ export function getBlockHierarchyRootClientId( state, clientId ) {
 	return parent;
 }
 
+/**
+ * Given a block client ID, returns the common ancestor with selected client ID.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which to find common ancestor client ID.
+ *
+ * @return {string} Common ancestor client ID or undefined
+ */
+export function getLowestCommonAncestorWithSelectedBlock( state, clientId ) {
+	const selectedId = getSelectedBlockClientId( state );
+	const clientParents = [ ...getBlockParents( state, clientId ), clientId ];
+	const selectedParents = [ ...getBlockParents( state, selectedId ), selectedId ];
+
+	let lowestCommonAncestor;
+
+	const maxDepth = Math.min( clientParents.length, selectedParents.length );
+	for ( let index = 0; index < maxDepth; index++ ) {
+		if ( clientParents[ index ] === selectedParents[ index ] ) {
+			lowestCommonAncestor = clientParents[ index ];
```

Comment:
Cool algorithm 😎

Reply:
Actually this was @koke solution to improve my original solution :) He helped me a lot during this styling logic.